pkg/build/buildkit/wait.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buildkit
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/moby/buildkit/client"
+	"github.com/okteto/okteto/pkg/env"
+	"github.com/okteto/okteto/pkg/log/io"
+)
+
+const (
+	// maxWaitTime is the maximum time to wait for buildkit to be available
+	maxWaitTime = 10 * time.Minute
+
+	// retryTime is the time to wait between retries
+	retryTime = 5 * time.Second
+
+	// maxBuildkitWaitTimeEnvVar is the environment variable to set the max wait time for buildkit
+	maxBuildkitWaitTimeEnvVar = "OKTETO_BUILDKIT_WAIT_TIMEOUT"
+
+	// retryBuildkitTimeEnvVar is the environment variable to set the retry time for buildkit
+	retryBuildkitTimeEnvVar = "OKTETO_RETRY_BUILDKIT_TIME"
+)
+
+// sleeper defines an interface for sleeping
+type sleeper interface {
+	Sleep(duration time.Duration)
+}
+
+// RealSleeper implements the Sleeper interface using time.Sleep
+type RealSleeper struct{}
+
+// Sleep sleeps for the specified duration
+func (s *RealSleeper) Sleep(duration time.Duration) {
+	time.Sleep(duration)
+}
+
+type buildkitClientFactoryWaiter interface {
+	GetBuildkitClient(ctx context.Context) (clientInfoRetriever, error)
+}
+
+type clientInfoRetriever interface {
+	Info(ctx context.Context) (*client.Info, error)
+}
+
+// buildkitWaiter encapsulates the logic to check if the BuildKit server is up and running
+type buildkitWaiter struct {
```

Comment:
Why do we have this type private to the package, but some of its properties are public? This sounds odd. Specially not using an interface in the consumer of this struct.

We are mixing approaches here. Why `waitingTime` is private, but we expose a function to retrieve its value, but for others, we do the opposite (make it public). 

Also, it sounds odd that `Logger` is exported. It should be internal and only log wait operations. I think it is even being used only internal.

Could you review this and follow the same pattern? I think this struct should also be public if it is going to be used directly outside this package